lib/SILOptimizer/Analysis/SideEffectAnalysis.cpp

```diff
@@ -347,10 +347,6 @@ void SideEffectAnalysis::analyzeInstruction(FunctionInfo *FInfo,
     case SILInstructionKind::ReleaseValueInst:
     case SILInstructionKind::UnownedReleaseInst:
       FInfo->FE.getEffectsOn(I->getOperand(0))->Releases = true;
-      
-      // TODO: Check the call graph to be less conservative about what
-      // destructors might be called.
-      FInfo->FE.setWorstEffects();
```

Comment:
I don't see how this is safe in general. Shouldn't the client have to say explicitly when they know a value hasn't escaped?

Reply:
FunctionEffects::getMemBehavior() returns the worst effects anyway if one of the parameter effects contains a "release".
The only thing which changed is that if a client explicitly looks at parameter effects, the client has to consider that a "released" parameter also potentially means worst effects on other parameters.

Reply:
Hm. It feels like it's too easy for someone to forget that, but I'll have to trust that the design makes sense for SideEffectAnalysis.

Reply:
Well, I should add a comment to the ABI functions